src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -771,13 +780,15 @@ string GetArchiveFileName (ITaskItem item)
 
 		private void AddNativeLibraries (ArchiveFileList files, string [] supportedAbis)
 		{
-			var frameworkLibs = FrameworkNativeLibraries.Select (v => new LibInfo {
-				Path = v.ItemSpec,
-				Link = v.GetMetadata ("Link"),
-				Abi = GetNativeLibraryAbi (v),
-				ArchiveFileName = GetArchiveFileName (v),
-				Item = v,
-			});
+			var frameworkLibs = FrameworkNativeLibraries
+				.Where (item => Path.GetFileName (item.ItemSpec) != "libarchive-dso-stub.so")
+				.Select (v => new LibInfo {
+					Path = v.ItemSpec,
+					Link = v.GetMetadata ("Link"),
+					Abi = GetNativeLibraryAbi (v),
+					ArchiveFileName = GetArchiveFileName (v),
+					Item = v,
+				});
```

Comment:
If `libarchive-dso-stub.so` was in an MSBuild pack, would it be easier to lookup the path? We could just look relative to the currently executing assembly?

Reply:
I've just had a thought that perhaps we should just place it in XABT.dll's resources.